locuszoom.app.js

```diff
@@ -267,6 +267,11 @@ LocusZoom.createCORSPromise = function (method, url, body, timeout) {
         };
         timeout && setTimeout(response.reject, timeout);
         body = typeof body !== "undefined" ? body : "";
+        // If posting an object set the proper content-type header
+        if (method == "POST"){
+            xhr.setRequestHeader("Content-Type","application/x-www-form-urlencoded");
```

Comment:
Adding `headers` to the argument list seems to make the most sense since `body` is already there. I can get that in place pretty quickly and that would push the header definition into `LocusZoom.Data.GeneConstraintSource.prototype.fetchRequest`, which is a much more appropriate place for that definition to live (as it's also where we're specifying it's a POST).

As for removing the JSON parsing I agree, the response should just be passed through to the resolve method on the promise and the default parse data function should execute the JSON parsing, allowing any custom data source to do something different if need be.